dask/array/utils.py

```diff
@@ -565,6 +589,32 @@ def svd_flip(u, v, u_based_decision=False):
     return u, v
 
 
+def matmul_safe(a, b):
+    xp = np
+
+    if is_cupy_type(a):
+        import cupy
+
+        xp = cupy
+
+    chunk = xp.matmul(a, b)
+    # Since we have performed the contraction via matmul
+    # but blockwise expects all dimensions back, we need
+    # to add one dummy dimension back
+    return chunk[..., xp.newaxis]
```

Comment:
I think this would be better left at its original place in `routines._matmul`, as this seems like a specialization from that function specifically. IOW, we only need `return xp.matmul(a, b)` here after NumPy/CuPy has been correctly chosen.

Reply:
@quasiben could you check my comment above?

Reply:
Sorry, I missed this @pentschev  .  I addressed this in https://github.com/dask/dask/pull/7563/commits/b13833c82eaf5ff470eff066ce1e427b91fd1001 .